src/test/java/io/pravega/connectors/flink/FlinkPravegaReaderTest.java

```diff
@@ -507,6 +508,31 @@ public void testGenerateUid() {
         assertNotEquals(uid1, uid3);
     }
 
+    @Test
+    public void testGenereateUidWithAutoGeneratedReaderGroupName() {
+        TestableStreamingReaderBuilder builder1 = new TestableStreamingReaderBuilder()
+                .withReaderGroupScope(SAMPLE_SCOPE)
+                .withReaderGroupName(GROUP_NAME)
+                .forStream(SAMPLE_STREAM, SAMPLE_CUT, StreamCut.UNBOUNDED);
+        String uid1 = builder1.generateUid();
+
+        TestableStreamingReaderBuilder builder2 = new TestableStreamingReaderBuilder()
+                .withReaderGroupScope(SAMPLE_SCOPE)
+                .withReaderGroupName("flink" + RandomStringUtils.randomAlphanumeric(20).toLowerCase())
```

Comment:
The test above have already covered the "random" name case because we are not specify the name. I think we don't need another test for this case. Instead, we should add a case with reader group name(just as `builder1 ` in this case) in the above test, and check if it's not equal to the others.

Reply:
This way of creating the `TestableStreamingReaderBuilder` doesn't create `readerGroupName` automatically, which means the `readerGroupName` is null. The test above doesn't cover the "random" name case. We don't specify the `readerGroupName` and the test case is ambiguous. I think merging the test case and the test case above is reasonable since specifying the `ReaderGrouName` is necessary in the test case.

Reply:
Yes, you are right, it's using the test builder.